src/common/MPPProber.cc

```diff
@@ -48,14 +48,22 @@ void MPPProber::run()
 {
     while (!stopped.load())
     {
-        std::this_thread::sleep_for(std::chrono::seconds(scan_interval));
+        {
+            std::unique_lock lock(scan_mu);
+            scan_cv.wait_for(lock, std::chrono::seconds(scan_interval), [this] () {
+                return stopped.load();
+            });
```

Comment:
if `pred() == false`, wait_for will enter sleep again rather than running the `scan()` in line 59. I think this is unexpected?